app/controllers/application_controller.rb

```diff
@@ -30,6 +30,11 @@ def add_breadcrumb(name, path = nil)
     breadcrumbs << Breadcrumb.new(name, path)
   end
 
+  # Render a 404 page for any undefined route
+  def render_not_found
+    render file: "#{Rails.root.join('public', '404.html')}, layout: false, status: :not_found
```

Comment:
I think the syntax error here is causing all the issues
```suggestion
    render file: "#{Rails.root.join('public', '404.html')}", layout: false, status: :not_found
```